bitsandbytes/cuda_setup/env_vars.py

```diff
@@ -56,16 +56,19 @@ def get_potentially_lib_path_containing_env_vars() -> Dict[str, str]:
 def restore_original_system_env_vars(env_vars: Dict[str, str]):
     """
         If the system supports `sudo` command, and the python program was started with `sudo`,
-        Use the `sudo -i env` command to obtain the original environment variables of the system, including CONDA_PREFIX, LD_LIBRARY_PATH...
+        Use the `sudo -i env` and `env` command to obtain the original environment variables of the system, including CONDA_PREFIX, LD_LIBRARY_PATH...
         Then replace the environment variables obtained from the current shell, to obtain correct and complete environment variables
     """
     if shutil.which('sudo') is None:
         return
     if os.geteuid() != 0:
         return
 
-    for env_line in subprocess.check_output(['sudo', '-i', 'env'], encoding="utf-8").splitlines():
-        if '=' in env_line:
-            key, value = env_line.split('=', 1)
-            if (key == "CONDA_PREFIX" or key == "LD_LIBRARY_PATH") and key not in env_vars:
-                env_vars[key] = value
+    sudo_env_output = subprocess.check_output(['sudo', '-i', 'env'], encoding="utf-8")
+    current_env_output = subprocess.check_output(['env'], encoding="utf-8")
```

Comment:
This is the same as just reading `os.environ`.

Reply:
![image](https://github.com/TimDettmers/bitsandbytes/assets/45415476/98e94dbe-e6a0-4197-8db8-e6b5b1e63cff)
if run with sudo, this is not the same as just reading os.environ.

Reply:
You're not testing for the right thing. 😁 

The environment variable `SUDO_COMMAND` includes the string `test_1` in your case, so that experiment will always return true:

```
# sudo python -c 'import subprocess;print("hello hello!");print(subprocess.check_output(["env"]))'
hello hello!
b'HOSTNAME=315f5cc030af\nLANG=C.UTF-8\nTERM=xterm\nPATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\nMAIL=/var/mail/root\nLOGNAME=root\nUSER=root\nHOME=/root\nSHELL=/bin/bash\nSUDO_COMMAND=/usr/local/bin/python -c import subprocess;print("hello hello!");print(subprocess.check_output(["env"]))\nSUDO_USER=root\nSUDO_UID=0\nSUDO_GID=0\n'
```
If I were to test whether `hello hello` appears in the output of `env`, it'd be true.